java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/WeightActivity.java

```diff
@@ -34,130 +34,117 @@
-    private void getWeights()
-    {
-        mHVClient.asyncRequest(
-                mCurrentRecord.getThingsAsync(ThingRequestGroup2.thingTypeQuery(Weight.ThingType)),
-    			new WeightCallback(WeightCallback.RenderWeights));
-    }
-    
-    private void putWeight(String value) {
-        final Thing2 thing = new Thing2();
+	private void putWeight(String value) {
+		final Thing2 thing = new Thing2();
 		thing.setData(new Weight(Double.parseDouble(value)));
-        // thing.setData(new Weight(-10));
-        mHVClient.asyncRequest(
-                mCurrentRecord.putThingAsync(thing),
-				new WeightCallback(WeightCallback.PutWeights));
-    }
-    
-    private void renderWeights(List<Thing2> things) {
-    	List<String> weights = new ArrayList<String>();
-        int count = 0;
-
-        for(Thing2 thing : things) {
-        	Weight w = (Weight)thing.getData();
-            if (count < 1)
-            {
-                TextView lastWeight = (TextView) findViewById(R.id.lasWeight);
-                lastWeight.setText(String.valueOf(w.getValue().getKg()));
-                count++;
-            }
-        	weights.add(String.valueOf(w.getWhen().getDate().getM() + "/" +
-                            w.getWhen().getDate().getD() + "/" +
-                            w.getWhen().getDate().getY()) + "               " +
-                            "                    " +
-                    String.valueOf(w.getValue().getKg()
-                    ));
-        }
-        ListView lv = (ListView)findViewById(R.id.weightList);
-        lv.setAdapter(new ArrayAdapter<String>(
-            WeightActivity.this,
-            android.R.layout.simple_list_item_1,
-            weights));
-    }
-    
-    public class WeightCallback<Object> implements RequestCallback {
-    	public final static int RenderWeights = 0;
-    	public final static int PutWeights = 1;
-    
-    	private int mEvent;
-    	
-    	public WeightCallback(int event) {
-            WeightActivity.this.setProgressBarIndeterminateVisibility(true);
-            mEvent = event;
-    	}
-
-        @Override
-        public void onError(HVException exception) {
-            WeightActivity.this.setProgressBarIndeterminateVisibility(false);
-            Toast.makeText(
-                WeightActivity.this, 
-                "An error occurred.  " + exception.getMessage(), 
-                Toast.LENGTH_LONG).show();
-        }
+		// thing.setData(new Weight(-10));
+		mHVClient.asyncRequest(mCurrentRecord.putThingAsync(thing),
+			new WeightCallback(WeightCallback.PutWeights));
+	}
+
+	private void renderWeights(List<Thing2> things) {
+		int count = 0;
+		for(Thing2 thing : things) {
+			Weight w = (Weight)thing.getData();
+		if (count < 1) {
+			TextView lastWeight = (TextView) findViewById(R.id.lasWeight);
+			lastWeight.setText(String.valueOf(w.getValue().getKg()));
+			count++;
+		}
+
+			mAdapter.add(String.valueOf(w.getWhen().getDate().getM() + "/" +
```

Comment:
Previous comment regarding naming isn't showing up so I'll add again. We should update the getM, getD, and getY methods to have better names (getMonth, getDay, and getYear)